MWPW-143904-New block: cc-mobile-app-banner

Ruchika4 opened the pull request:
* New block to allow authors to add branch io app banner
[Authoring documentation](https://stage--cc--adobecom.hlx.page/docs/authoring/cc-mobile-app-banner?martech=off)

Resolves: [MWPW-143904](https://jira.corp.adobe.com/browse/MWPW-143904)

**Test URLs:**
- Before: https://stage--cc--adobecom.hlx.live/drafts/ruchika/app-banner/photoshop-lightroom3?martech=off
- After: https://mobile-app-banner--cc--adobecom.hlx.live/drafts/ruchika/app-banner/photoshop-lightroom3?martech=off

aishwaryamathuria commented:
@Ruchika4 
1. The branch has reduced code coverage compared to the current stage. Could you pull the latest code and check the same?
2. The PSI check is failing. Could you add a valid .live url?
CC: @salonijain3

aishwaryamathuria commented on creativecloud/blocks/cc-mobile-app-banner/cc-mobile-app-banner.js:
```diff
@@ -0,0 +1,69 @@
+  const config = getConfig();
+  let keyMatch = [];
+  const resp = await fetch(`${config.contentRoot ?? ''}/branch-io-key.json`);
+  if (resp.ok) {
```
if response return is not ok we are going to get an erro rin the later return `return keyMatch[0]?.key;`
Can we do an early return here? 
if (!resp.ok) return null;

Ruchika4 replied:
I think it'll return undefined if the resp is not ok and init method we handle. So I believe that it will get handled

aishwaryamathuria commented on creativecloud/blocks/cc-mobile-app-banner/cc-mobile-app-banner.js:
```diff
@@ -0,0 +1,69 @@
+function branchInit(header, key) {
+  let initValue = false;
+  function initBranch() {
+    if (initValue) {
```
```suggestion
    if (initValue) return;
```

aishwaryamathuria commented on creativecloud/blocks/cc-mobile-app-banner/cc-mobile-app-banner.js:
```diff
@@ -0,0 +1,69 @@
+  const header = document.querySelector('.global-navigation');
+  if (!header) return;
+  const classListArray = Array.from(el.classList);
+  const product = classListArray.find((token) => token.startsWith('product-')).split('-')[1];
```
```suggestion
  const product = [...el.classList].find((token) => token.startsWith('product-'))?.split('-')[1];
```

Ruchika4 commented:
> @Ruchika4
> 
> 1. The branch has reduced code coverage compared to the current stage. Could you pull the latest code and check the same?
> 2. The PSI check is failing. Could you add a valid .live url?
> 
> CC: @salonijain3

Hi @aishwaryamathuria ,

1. Because of third party integration and milo blocking loading of 3rd party scriptin unit test hence full test coverage could not be achieved. Because of the urgency I have just moved the block from milo to CC. As CC repo I believe doesn't block the 3rd party script in unit test case I'll work on adding more unit test cases later.
2. Updated the url to .live. PSI is also checking authoring documentation url. I believe we can ignore that.

spadmasa commented:
Validation done in devices on the below url :
https://mobile-app-banner--cc--adobecom.hlx.live/drafts/ruchika/app-banner/photoshop-lightroom3?martech=off
![image](https://github.com/adobecom/cc/assets/73129464/a79714e5-72ef-46f1-a5cb-b9cdc3b57549)
![image](https://github.com/adobecom/cc/assets/73129464/5d9f2748-533d-4b39-b446-a7f21c8d793b)

![image](https://github.com/adobecom/cc/assets/73129464/25174857-e152-4eb2-9211-0548646b4e82)

![image](https://github.com/adobecom/cc/assets/73129464/6dfdf153-24a3-4b7d-930e-9824b178bdeb)
![image](https://github.com/adobecom/cc/assets/73129464/54c53dda-f12a-47b8-b5f3-bb6a9006b010)

![image](https://github.com/adobecom/cc/assets/73129464/320a42b8-df4a-46e7-b669-d22a327ccce8)

![image](https://github.com/adobecom/cc/assets/73129464/e9aa62f8-cff4-4669-bc00-a311cfcf17eb)

![image](https://github.com/adobecom/cc/assets/73129464/5e3cefcb-d8c2-4b67-950c-e10936ea5c88)
